src/poetry/console/commands/env/activate.py

```diff
@@ -41,30 +41,39 @@ def handle(self) -> int:
         )
 
     def _get_activate_command(self, env: Env, shell: str) -> str:
-        if shell == "fish":
-            command, filename = "source", "activate.fish"
-        elif shell == "nu":
-            command, filename = "overlay use", "activate.nu"
-        elif shell in ["csh", "tcsh"]:
-            command, filename = "source", "activate.csh"
-        elif shell in ["powershell", "pwsh"]:
-            command, filename = ".", "activate.ps1"
+        shell_configs = {
+            "fish": ("source", "activate.fish"),
+            "nu": ("overlay use", "activate.nu"),
+            "csh": ("source", "activate.csh"),
+            "tcsh": ("source", "activate.csh"),
+            "powershell": (".", "activate.ps1"),
+            "pwsh": (".", "activate.ps1"),
+            "cmd": (".", "activate.bat"),
+        }
+
+        command, filename = shell_configs.get(shell, ("source", "activate"))
+
+        activation_script = env.bin_dir / filename
+
+        if not activation_script.exists():
+            if shell == "cmd" and not WINDOWS:
+                fallback_script = env.bin_dir / "activate"
+                if fallback_script.exists():
+                    return f"source {self._quote(str(fallback_script), 'bash')}"
```

Comment:
What is this for? I know that Powershell exists for Linux but cmd? Does this cover a real-world use case? Does a test fail if we remove this?